### PR TITLE
ScreenShot feature

### PR DESCRIPTION
**Pull request by psav**
Doesn't rely on rbac any more

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit ad4cb6efc9d24483b958b7e16c872bbe94a3200e:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/fixtures/pytest_selenium.py:38:10: E401 multiple imports on one line
./cfme/fixtures/pytest_selenium.py:41:1: F401 'art_client' imported but unused
```
:trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 929cde063f446e37d14a4bdd1f03947b356f94a8:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 3695d90cf535f02f9866463bb6e11116c2acfb1f:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4808a21ade1930e4265ce3acc69cbe314341658f:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by psav** [comment]
This can be made better......indeed, maybe I should just wait a bit before merging......

**Comment by dajoRH** [comment]
Lint report for commit fd31f726821d6627072a31d7b88995e8a986c6da:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/fixtures/pytest_selenium.py [hunk 1]
```diff
@@ -319,6 +326,24 @@ def _nothing_in_flight():
...
+    url = browser().current_url
+    url = url.replace(base_url(), '')
+    url = url.replace("/", '_')
+    if url not in urls:
+        logger.info('Taking picture of page: {}'.format(url))
+        ss, sse = take_screenshot()
+        if ss:
+            ss_path = log_path.join('page_screenshots')
+            if not os.path.exists(ss_path.strpath):
+                os.mkdir(ss_path.strpath)
```
Comment (mfalesni) [inline cfme/fixtures/pytest_selenium.py]: Why you use `os` features when you can just do `ss_path.exists()` and ``ss_path.mkdir()`` ?

File: cfme/fixtures/pytest_selenium.py [hunk 2]
```diff
@@ -319,6 +326,24 @@ def _nothing_in_flight():
...
+            with open(ss_path.join("{}.png".format(url)).strpath, 'wb') as f:
```
Comment (mfalesni) [inline cfme/fixtures/pytest_selenium.py]: `with ss_path.join(...).open('wb') as f:`

File: cfme/fixtures/pytest_selenium.py (hunk 1 above)
Comment (psav) [inline cfme/fixtures/pytest_selenium.py]: Because I didn't know they could do that :p

File: cfme/fixtures/pytest_selenium.py (hunk 2 above)
Comment (psav) [inline cfme/fixtures/pytest_selenium.py]: Any functional reason why this is different?

Comment (mfalesni) [inline cfme/fixtures/pytest_selenium.py]: It is the same with regard to functionality, but it seems logical to me to use facilities provided by `py.path`. And more readable.

**Comment by dajoRH** [comment]
Lint report for commit c29502e035ef83e8fced99c7f34c52796e075a27:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by seandst** [comment]
Is this wired into the artifactor?

If not, can it be? :)

**Pull request merged**